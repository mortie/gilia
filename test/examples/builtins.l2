print "+"
print +()
print +(10)
print 10 + 20
print (+ 10 20 30)

print "\n-"
print -() -(10)
print 10 - 20
print (- 10 20 30)

print "\n/"
print /() /(10)
print 10 / 2
print (/ 10 2 2)

print "\n*"
print *() *(10)
print 10 * 2
print (* 10 20 30)

print "\n=="
print ==() ==(10)
print 10 == 10 20 == 10
print (== 10 20 30)
print (== 'a 'a 'a)

print "\n!="
print !=() !=(10)
print 10 != 10 20 != 10
print (!= 10 20 30)
print (!= 'a 'a 'a)

print "\n>"
print 10 > 20
print 20 > 10
print 10 > 10
print (> 1 10 100 200)

print "\n>="
print 10 >= 20
print 20 >= 10
print 10 >= 10
print (>= 1 10 100 200)

print "\n<"
print 10 < 20
print 20 < 10
print 10 < 10
print (< 1 10 100 200)

print "\n<="
print 10 <= 20
print 20 <= 10
print 10 <= 10
print (<= 1 10 100 200)

print "\nprint"
print none
print 'true 'false 'hello
print 100
print 100.5
print 0xff
print "Hello World"
print [none 'true 'false 'hello 100.1 "Nope" {} {0}]
print {} {a: 10}

print "\nlen"
print (len 10)
print (len [])
print (len [10 20])
print (len "Hello")
print (len {a: 10; b: 20; c: 30})

print "\n&&"
print &&()
print &&('true)
print &&('false)
print 'true && 'true
print (&& 10 == 10 20 == 20 30 != 40)
print 'true && 10

print "\n||"
print ||()
print ||('true)
print ||('false)
print 'true || 'true
print 'true || 'false
print (|| 'false 10 'true)
print (|| 'false 10 'nope)

print "\n??"
print none ?? none ?? 30 ?? 20
print (?? 10 none none)
print (?? none none "hello")
print ??()
