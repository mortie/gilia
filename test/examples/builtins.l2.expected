+
0
10
30
60

-
0 -10
-10
-40

/
1 0.1
5
2.5

*
1 10
20
6000

==
(true) (true)
(true) (false)
(false)
(true)

!=
(false) (false)
(false) (true)
(true)
(false)

>
(false)
(true)
(false)
(false)

>=
(false)
(true)
(true)
(false)

<
(true)
(false)
(false)
(true)

<=
(true)
(false)
(true)
(true)

print
(none)
(true) (false) (atom 23)
100
100.5
255
Hello World
[(none) (true) (false) (atom 23) 100.1 Nope (namespace) (function)]
(namespace) (namespace)

len
0
0
2
5
3

&&
(true)
(true)
(false)
(true)
(true)
(false)

||
(false)
(true)
(false)
(true)
(true)
(true)
(false)

??
30
10
hello
(none)
